src/praisonai-agents/praisonaiagents/knowledge/knowledge.py

```
@@ -686,7 +686,17 @@ def index(
                     result.files_skipped += 1
                     continue
-                
+
+                # Remove the previous version's chunks before re-adding a changed
+                # file, otherwise the vector store grows unbounded and stale,
+                # now-incorrect chunks stay retrievable alongside current content.
+                if incremental:
+                    for old_id in tracker.get_memory_ids(filepath):
+                        try:
+                            self.delete(old_id)
+                        except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)